django/db/models/base.py

```diff
@@ -75,6 +75,18 @@ def __new__(cls, name, bases, attrs, **kwargs):
         classcell = attrs.pop('__classcell__', None)
         if classcell is not None:
             new_attrs['__classcell__'] = classcell
+        # Fix related to bug #28198.
+        # Remove duplicate public attributes from parent classes
+        attr_per_class = {}
+        for parent in parents:
+            attr_per_parent = {}
+            for key in attrs.keys():
```

Comment:
`.keys()` can be removed throughout ala 21046e77734278cea871dce922220bf29aa5b7b4.